nixos/modules/services/development/jupyterhub/default.nix

```diff
@@ -0,0 +1,117 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.jupyterhub;
+
+  jupyterhub_env = pkgs.python3.withPackages (p: with p; cfg.hubPackages);
+  jupyterlab_env = pkgs.python3.withPackages (p: with p; cfg.jupyterPackages);
```

Comment:
you'll probably want to parameterize the python interpreter, as you could have someone passing packages of a different python interpreter version to this

Reply:
done. I've done this via `jupyterhubEnv`, `jupyterlabEnv`, and `kernels` options.